### ISSUE TITLE
Debug mode - error

### ISSUE THREAD

**Issue by BigMac80** [MSEndpointMgr/ModernDriverManagement#216]
I have setup MDM following the user guide and steps, and I am attempting to validate the driver packages but I am getting the same error message that I am seeing when i try to run it from the task sequence - Error **InnerTerminatingFailure**

Here is the command I am using in the task sequence.

.\Invoke-CMApplyDriverPackage.ps1 -BareMetal -Endpoint 'PCName' -TargetOSName 'Windows 10' -TargetOSVersion '21H2'

If anyone could assist that would be greatly appreciated.

Invoke-CMApplyDriverPackage.ps1 : **InnerTerminatingFailure**
At line:1 char:1
+ .\Invoke-CMApplyDriverPackage.ps1 -DebugMode -Endpoint 
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotImplemented: (:String) [Invoke-CMApplyDriverPackage.ps1], RuntimeException
    + FullyQualifiedErrorId : Invoke-CMApplyDriverPackage.ps1

![2022-12-14 16_13_38-Error](https://user-images.githubusercontent.com/109587137/207650132-97d3c3f0-0d9d-47fe-bbd0-97bb767a269a.png)

**Comment by stephannn** [comment]
Hi,
my colleague told me that happens if some parameter may be missing. He was for example requesting some driver package that did not exist for the OS. Maybe that helps a bit.

**Comment by shanu77122** [comment]
As stephannn said, only to find what is missing, open the ApplyDriverPackage.log which will be sitting in the CCM\Logs folder of the client if you're using MECM

**Comment by BigMac80** [comment]
Thanks for the update. Currently I have one package deployed for a latitude 5430 model as I have only just started using MDM but will check the logs again.
As far as I can tell everything is configured correctly so not sure why I am seeing this error.
Sent from Yahoo Mail on Android 
 
  On Thu, 12 Jan 2023 at 16:00, ***@***.***> wrote:   


As stephannn said, only to find what is missing, open the ApplyDriverPackage.log which will be sitting in the CCM\Logs folder of the client if you're using MECM

**Comment by BigMac80** [comment]
I have checked the AppyDriverPackage.log file and I am seeing the following error:

 - Failed to download package content with PackageID 'PR1002B4'. Return code was: -2147467259
  InnerTerminatingFailure	ApplyDriverPackage
[ApplyDriverPackage]: Apply Driver Package process failed, please refer to previous error or warning messages

Has anyone come across this error before.

**Comment by stephannn** [comment]
Not sure, but it does not seem to be an error with the script because the packageID is already transmitted to the SCCM client:
`$ReturnCode` = Invoke-Executable -FilePath (Join-Path -Path $env:windir -ChildPath "CCM\OSDDownloadContent.exe")

Does the package exist on the distribution point and does it work on another system?

**Comment by BigMac80** [comment]
Yes I can confirm that the package is ditributed to the DP and I cannot get it to work on any of our devices.

**Comment by stephannn** [comment]
What the script basically does is checking which is the correct package and writing it in the TS variable:
`$TSEnvironment.Value("OSDDownloadDownloadPackages") = "$($PackageID)"`

after that it is trying download it:
_$ReturnCode = Invoke-Executable -FilePath (Join-Path -Path $env:windir -ChildPath "CCM\OSDDownloadContent.exe")_

can you execute these commands manually and also maybe try another package? Be aware, you have to be in a TS

**Comment by BigMac80** [comment]
I have tried multiple packages and they result in the same error message Invoke-CMApplyDriverPackage.ps1 : **InnerTerminatingFailure**.

I have tried running the commands manually and this is what I am seeing:

![image](https://user-images.githubusercontent.com/109587137/223102483-b4271d3f-1b56-4ee2-a1cd-451b3d2a430c.png)

And here is the error in the ApplyDriverPackage.log:

![image](https://user-images.githubusercontent.com/109587137/223102709-ef97a1f2-f9c8-445b-b9e4-a3aeda74180a.png)

Is there anything special that needs to be set on the package itself in terms of access i.e. does the following need to be enabled.

![image](https://user-images.githubusercontent.com/109587137/223090392-819964ee-1ee0-4bdd-9618-c864f97415dc.png)

![image](https://user-images.githubusercontent.com/109587137/223090483-dcf6da53-68ee-4765-9ca6-ad5d93c00788.png)

Is there anything else that could be missing.

**Comment by Latiron** [comment]
I have the same error message from the ApplyDriverPackage.log file

 - Failed to download package content with PackageID 'H01002DF'. Return code was: -2147221246	ApplyDriverPackage	5/11/2023 11:27:22 AM	1644 (0x066C)
InnerTerminatingFailure	ApplyDriverPackage	5/11/2023 11:27:22 AM	1644 (0x066C)
[ApplyDriverPackage]: Apply Driver Package process failed, please refer to previous error or warning messages	ApplyDriverPackage	5/11/2023 11:27:22 AM	1644 (0x066C)

However, it's only happened for this new Dell Latitude 7440 model.  I just started using this product, and can confirm from all my testing this week, that it is working successfully on the following models I have tested on so far:
Dell Latitude 7420
Dell Latitude 7430
Dell Precision 5820
Dell Optiplex 7000
Microsoft Surface Book 3 1899

This Dell Latitude 7440 is the first one to do this.  To troubleshoot the issue, I have deleted the package from the Primary DP where the imaging is taking place from.  I then deleted the Latitude 7440 package that was created by the Automated Driver Tool, opened that tool back up and re-ran the process to have it created again.  No need to distribute it from MECM as the tool does that for me.

The imaging process was started again and failed at the same spot with the same message for this DriverPackage.

I know this is a fairly new Dell model, but I wouldn't think any of that would be the problem since the tool found the driver and created the package successfully.  So I am stumped as to why the DriverPackage is unable to be downloaded to the computer during this process.

If *anyone* has a clue on this, I'm all ears.  My support folks got in 7 of these computers, and I need to get this fixed asap.  Otherwise I will end up having to create a one-off Task Sequence just so they can image these particular computers.

**Comment by stephannn** [comment]
@Latiron , one question. Can you download the driver due the Task Sequence as a package? Maybe the SMSTS gives out another type of error like Hash error

**Comment by Latiron** [comment]
I can certainly try that.  But please note, I am very green at all of this, so if you happen to have any steps to do what you are mentioning here, I would very much appreciate the assist.

**Comment by stephannn** [comment]
@Latiron , I am pretty sure the error has nothing to do with the Driver Management Tool because it is "just" triggering SCCM functions. In special when you say it is not happening to all driver packages. So it could be related to AntiVirus (MS recommends exclusions) because sometimes it changing some bytes that results in a different hash value.

**Comment by TankNation** [comment]
@BigMac80, did you ever find a resolution to what was causing
`Failed to download package content with PackageID 'xxxxxxxx'. Return code was: -2147467259`
